--- ./devices.c.orig	2015-07-28 13:04:57.476014412 +0300
+++ ./devices.c	2015-07-28 13:14:39.608024101 +0300
@@ -27,8 +27,6 @@
 
 #if (HOST_OS != SOLARIS)
 #include "libusb-utils.h"
-#undef HAVE_IEEE1284_H
-#include <ieee1284.h>
 #else
 
 int perform_device_id_ioctl(int port, const char *device_file, 
@@ -51,8 +49,6 @@
     switch (port) {
     case USB:
       return DEV_USB_LP_INACCESSIBLE;
-    case PARPORT:
-      return DEV_LP_INACCESSIBLE;
     }
   }
 
@@ -87,76 +83,13 @@
 int get_device_id(const int port, const char *device_file, 
                   const int portnumber, char *device_id) {
 
-#if (HOST_OS != SOLARIS)
-  struct parport_list parports;
-#endif
-
   char tmp[BUFLEN];
   char device_file1[256];
   int size;
   int fd;
   int result = COULD_NOT_GET_DEVICE_ID;
 
-  if (port == PARPORT ) {
-
-#if (HOST_OS != SOLARIS)
-
-    /* check if we have appropriate permissions */
-
-#if (HOST_OS == LINUX)
-
-    sprintf(device_file1, "/dev/parport%d", portnumber);
-
-    if ((fd = open(device_file1, O_RDWR)) < 0) {
-      return DEV_PARPORT_INACCESSIBLE;
-    }
-
-    close(fd);
-
-    sprintf(device_file1, "/dev/lp%d", portnumber);
-    
-    if ((fd = open(device_file1, O_RDWR)) < 0) {
-      return DEV_LP_INACCESSIBLE;
-    }
-
-    close(fd);
-
-#elif (HOST_OS == FREEBSD)
-
-    sprintf(device_file1, "/dev/lpt%d", portnumber);
-    
-    if ((fd = open(device_file1, O_RDWR)) < 0) {
-      return DEV_LP_INACCESSIBLE;
-    }
-
-    close(fd);
-
-#endif
-
-    /* for linux and freebsd use libieee1284 to get the device id from 
-       parallel port */
-
-    if (ieee1284_find_ports(&parports, 0) == E1284_OK) {
-      if (portnumber < parports.portc) {
-	size = ieee1284_get_deviceid(parports.portv[portnumber], -1, 
-                                     F1284_FRESH, tmp, BUFLEN);
-	if (size >= 2) {
-	  strncpy(device_id, tmp + 2, size - 2);
-	  return OK;
-	}
-      }
-    }
-
-    return COULD_NOT_GET_DEVICE_ID;
-
-#else
-    
-    /* solaris */
-    return perform_device_id_ioctl(port, "/dev/ecpp%d", portnumber, device_id);
-
-#endif
-    
-  } else if (port == USB) {
+  if (port == USB) {
 
 #if (HOST_OS != SOLARIS)
 
@@ -211,11 +144,7 @@
 
 #if (HOST_OS == LINUX)
 
-  if (port == PARPORT) {
-    sprintf(device_file, "/dev/lp%d", portnumber);
-  } else {
-    return UNKNOWN_PORT_SPECIFIED;
-  }
+  return UNKNOWN_PORT_SPECIFIED;
 
 #elif (HOST_OS == FREEBSD) 
 
