--- a/configure
+++ b/configure
@@ -547,13 +547,11 @@
 if [ "$otr" = 1 ]; then
 	# BI == built-in
 	echo '#define OTR_BI' >> config.h
-	echo "EFLAGS+=-L${otrprefix}/lib -lotr" >> Makefile.settings
-	echo "CFLAGS+=-I${otrprefix}/include" >> Makefile.settings
+	echo "EFLAGS+=-lotr" >> Makefile.settings
 	echo 'OTR_BI=otr.o' >> Makefile.settings
 elif [ "$otr" = "plugin" ]; then
 	echo '#define OTR_PI' >> config.h
-	echo "OTRFLAGS=-L${otrprefix}/lib -lotr" >> Makefile.settings
-	echo "CFLAGS+=-I${otrprefix}/include" >> Makefile.settings
+	echo "OTRFLAGS=-lotr" >> Makefile.settings
 	echo 'OTR_PI=otr.so' >> Makefile.settings
 fi
 
